### DISCUSSION TITLE
Invisible glowing block entity

### DISCUSSION THREAD

**Discussion by UnfamousThomas** [Minestom/Minestom#2262] [Help] [answered]
Hello.

I am trying to figure out how to make a block display entity, that is invisible but glowing and a player can pass through it.

First tried with Minestom, like this:

```
    public TestDisplayEntity(@NotNull UUID uuid) {
        super(EntityType.BLOCK_DISPLAY, uuid);

        BlockDisplayMeta blockDisplayMeta = (BlockDisplayMeta) this.getEntityMeta();

        blockDisplayMeta.setHasNoGravity(true);
        blockDisplayMeta.setScale(new Vec(100, 5, 100));
        blockDisplayMeta.setHasGlowingEffect(true);
        blockDisplayMeta.setInvisible(true);
}    
```

but then could not figure out what was wrong, so tries with pure Minecraft commands too, e.g:
`/summon block_display ~ ~1 ~ {Glowing:1b}`

and that did not work. Am I missing something super obvious here?

**Comment by cosrnic** [comment]
You have to set the block state, try `blockDisplayMeta.setBlockState(Block.GRASS_BLOCK)`, in vanilla i think you have to add `block_state:{Name:"minecraft:grass_block"}` to the nbt

**Reply by UnfamousThomas** [reply]
Hmm, yeah that works, but I failed to clarify it a bit. The ideal solution would be one you can passthrough as a player... Which is why I hoped keeping on default would work but looks like not.

**Reply by cosrnic** [reply, marked as answer]
Display entities have no collision to the player so you should be able to pass through them although you might have to use a glass block if you want it to be see-through, i tried a barrier but it didn't render the glowing effect for me. if you are using a resource pack you might be able to use an item display with an invisible texture and change its model to look like a block

**Reply by UnfamousThomas** [reply]
Yeah, you are right. Must be some weird default Minecraft thing that decides to not render invisible block displays.

Anyways, thanks for the help.
